File: Python/1. Greedy/Q1_bigN.cpp
#include <iostream>
#include <algorithm>

using namespace std;

int main(){
  int N, M, K;
  cin >> N >> M >> K;
  int numbers[N+1];
  int ans = 0;

  for(int i = 0; i < N; i++){
    cin >> numbers[i];

  }

  sort(numbers, numbers+N);

  for(int i = 0; i < M; i++){
    if((i+1) % K == 0){
      ans += numbers[N-2];
    }else{
      ans += numbers[N-1];
    }
  }

  cout << ans << endl;
}

File: Python/1. Greedy/Q1_bigN.py
n,m,k = map(int, input().split())

data = list(map(int, input().split()))

data.sort()

first = data[n-1]
second = data[n-2]

# 가장 큰 수가 더해지는 갯수 
count = int(m / (k+1)) * k
count += m % (k+1)

result = 0
result += (count) * first
result += (m-count) * second

print(result)

# ans = 0
#
# for i in range(m):
#     if (i+1) % k == 0:
#         ans += second
#     else:
#         ans += first
#
# print(ans)
